hazelcast/src/main/java/com/hazelcast/query/PagingPredicate.java

```diff
@@ -71,16 +70,13 @@
  */
 public class PagingPredicate implements IndexAwarePredicate, DataSerializable {
 
-    private Predicate predicate;
+    private static final Map.Entry<Integer, Map.Entry> NULL_ANCHOR = new SimpleImmutableEntry(-1, null);
 
+    private final LinkedList<Map.Entry<Integer, Map.Entry>> anchorList = new LinkedList<Map.Entry<Integer, Map.Entry>>();
```

Comment:
there is an assumption that on Xth position is an anchor for page X. Is this correct? Are you sure this assumption is always valid? the code is quite complex, it's not obvious too me.